src/resmom/mom_hook_func.c

```diff
@@ -3002,6 +3002,24 @@ mom_process_hooks(unsigned int hook_event, char *req_user, char *req_host,
 								at_flags |=
 				(ATR_VFLAG_SET | ATR_VFLAG_MODCACHE);
 			set_job_exit = 1;
+		} else if ((hook_event == HOOK_EVENT_EXECJOB_LAUNCH) && (num_run >= 1)) {
+			/*
+			 * If there are multiple execjob_launch hooks, we need to cascade
+			 * the execjob_launch specific parameters to the next execjob_launch
+			 * hook, if any of the previous hooks has set these values.
+			 */
+			if (hook_output != NULL) {
+
+			  if (hook_output->progname != NULL) {
+				  hook_input->progname = *hook_output->progname;
```

Comment:
- Indentation is a little off at this place.
- can there be case that hook_input->progname, env and argv are already pointing to memory from heap? do we need to free that before assignment?